pyhf/writexml.py

```diff
@@ -1,45 +1,207 @@
+import logging
+
 import os
 import xml.etree.cElementTree as ET
+import numpy as np
+import uproot
+from uproot_methods.classes import TH1
+
+_ROOT_DATA_FILE = {}
+_HISTNAME = "h{sample}{modifier}{highlow}_{channel}_obs_cuts"
+
+log = logging.getLogger(__name__)
+
+
+def export_root_histogram(histname, data):
+    h = TH1.from_numpy((np.asarray(data), np.arange(len(data) + 1)))
+    h._fName = histname
+    # NB: uproot crashes for some reason, figure out why later
+    # if histname in _ROOT_DATA_FILE:
+    #    raise KeyError('Duplicate key {0} being written.'.format(histname))
+    _ROOT_DATA_FILE[histname] = h
 
 
-def measurement(lumi, lumierr, poi, param_settings=None, name='Meas1'):
-    param_settings = param_settings or []
+# https://stackoverflow.com/a/4590052
+def indent(elem, level=0):
+    i = "\n" + level * "  "
+    if len(elem):
```

Comment:
Can we use the [CodeFactor suggestion](https://www.codefactor.io/repository/github/diana-hep/pyhf/pull/426) here RE: "Do not use `len(SEQUENCE)` to determine if a sequence is empty"?

Reply:
Thanks! :slightly_smiling_face: